features/steps/component.go

```diff
@@ -2,235 +2,100 @@ package steps
-	GitCommit string = "7434fe334d9f51b7239f978094ea29d10ac33b16"
-	Version   string = ""
+	BuildTime = strconv.Itoa(time.Now().Nanosecond())
+	GitCommit = "component test commit"
+	Version   = "component test version"
 )
 
 type Component struct {
 	componenttest.ErrorFeature
-	ApiFeature        *componenttest.APIFeature
-	errorChan         chan error
-	DatasetAPI        *httpfake.HTTPFake
-	CantabularClient  *mock.CantabularClient
-	svc               *service.Service
-	cfg               *config.Config
-	wg                *sync.WaitGroup
-	signals           chan os.Signal
-	ctx               context.Context
-	HTTPServer        *http.Server
-	store             service.Datastore
-	g                 service.Generator
-	shutdownInitiated bool
-	consumer          kafka.IConsumerGroup
-	waitEventTimeout  time.Duration
+	AuthFeature       *componenttest.AuthorizationFeature
+	APIFeature        *componenttest.APIFeature
+	MongoFeature      *MongoFeature
+	DatasetFeature    *DatasetFeature
+	CantabularFeature *CantabularFeature
+
+	svc *service.Service
 }
 
-func NewComponent(t *testing.T, zebedeeURL, mongoAddr string) (*Component, error) {
-	ctx := context.Background()
+func NewComponent(t *testing.T) *Component {
+	g := &mock.Generator{URLHost: "http://mockhost:9999"}
 	cfg, err := config.Get()
 	if err != nil {
-		return nil, fmt.Errorf("failed to get config: %w", err)
-	}
-
-	cfg.ZebedeeURL = zebedeeURL
-	cfg.Mongo.ClusterEndpoint = mongoAddr
-	cfg.Mongo.Database = utils.RandomDatabase()
-
-	g := &mock.Generator{
-		URLHost: "http://mockhost:9999",
+		t.Fatalf("failed to get config: %s", err)
 	}
 
-	mongoClient, err := GetWorkingMongo(ctx, cfg, g)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create new mongo mongoClient: %w", err)
+	component := &Component{
+		ErrorFeature:      componenttest.ErrorFeature{TB: t},
+		AuthFeature:       componenttest.NewAuthorizationFeature(),
+		DatasetFeature:    NewDatasetFeature(t).Init(cfg),
+		CantabularFeature: NewCantabularFeature().Init(),
 	}
+	component.MongoFeature = NewMongoFeature(component.ErrorFeature, g, cfg).Init()
+	component.APIFeature = componenttest.NewAPIFeature(component.Router)
 
-	return &Component{
-		errorChan:  make(chan error),
-		wg:         &sync.WaitGroup{},
-		ctx:        ctx,
-		HTTPServer: &http.Server{},
-		cfg:        cfg,
-		DatasetAPI: httpfake.New(httpfake.WithTesting(t)),
-		CantabularClient: &mock.CantabularClient{
-			OptionsHappy: true,
-		},
-		store:            mongoClient,
-		g:                g,
-		waitEventTimeout: WaitEventTimeout,
-	}, nil
-}
-
-// Init initialises the server, the mocks and waits for the dependencies to be ready
-func (c *Component) Init() (http.Handler, error) {
-	c.signals = make(chan os.Signal, 1)
-	signal.Notify(c.signals, os.Interrupt, syscall.SIGTERM)
+	cfg.ZebedeeURL = component.AuthFeature.FakeAuthService.ResolveURL("")
+	log.Info(context.Background(), "config used by component tests", log.Data{"cfg": cfg})
```

Comment:
Could we pass the relevant context to use here rather than background()?

Reply:
I don't think we have a relevant context at this point? To be honest I'd prefer to get rid of the log altogether - it may be okay while debugging the tests but the tests should not need to be printing/logging - would you be okay to take it out?